### PR TITLE
If ContextCompat is available, use it to resolve drawables

### LINKED ISSUE #1422: Use ContextCompat to retrieve drawables
Given I'm injecting a drawable into my fragment like that:

```
@DrawableRes(R.drawable.bg_button_with_stroke_orange)
protected Drawable backgroundDrawable;
```

Then Android Annotations generates this code:

```
private void init_() {
    Resources resources_ = context_.getResources();
    backgroundDrawable = resources_.getDrawable(drawable.bg_button_with_stroke_orange);
}
```

Which then generates this compiler warning for `targetSdk=22` builds:

```
TheClass_.java:33: warning: [deprecation] getDrawable(int) in Resources has been deprecated
        backgroundDrawable = resources_.getDrawable(drawable.bg_button_with_stroke_orange);
```

Instead, AndroidAnnoations should, given the class [ContextCompat](https://developer.android.com/reference/android/support/v4/content/ContextCompat.html) is available, generate this:

```
private void init_() {
    backgroundDrawable = ContextCompat.getDrawable(context_, drawable.bg_button_with_stroke_orange);
}
```

Benefits:
* Obviously, the compiler warning would be gone
* More importantly, the theme will be applied to the drawable on Lollipop+ devices

### PR DESCRIPTION
**Pull request by ened**
Tries to fix #1422.

### CODE REVIEW

**Comment by WonderCsabo** [comment]
Thanks! Can you squash the commits? Also please add two tests in a separate commit: one when ContextCompat is available and one when it is not. Then assert the generated code is correct.  You can check out ActivityIntentFragmentTest for an example.

**Comment by ened** [comment]
@WonderCsabo please take a look.

File: AndroidAnnotations/androidannotations/src/test/java/org/androidannotations/generation/ContextCompatDrawableTest.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	private static final String DRAWABLE_VIA_SUPPORT_SIGNATURE = ".*myDrawable = ContextCompat\\.getDrawable\\(this, drawable\\.myDrawable\\);.*";
+
+	@Before
+	public void setUp() {
+		addProcessor(AndroidAnnotationProcessor.class);
+	}
+
+	@Test
+	public void activityCompilesWithRegularDrawable() {
+		addManifestProcessorParameter(ContextCompatDrawableTest.class, "AndroidManifestForDrawable.xml");
```
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations/src/test/java/org/androidannotations/generation/ContextCompatDrawableTest.java]: You can move this line to `setUp` from both test methods.

**Comment by WonderCsabo** [comment]
Nice PR, thanks!

**Pull request merged**